src/gfn/modules.py

```diff
@@ -90,32 +95,11 @@ def forward(self, input: States | torch.Tensor) -> torch.Tensor:
         """
         if isinstance(input, States):
             input = self.preprocessor(input)
-
-        out = self.module(input)
```

Comment:
It's a shame to remove all this from the base class and have to re-implement it in the all of the classes that inherit from this one.

Is this only because of `self._output_dim_is_checked` or is there more?

Reply:
Yes, I am not checking output here, but in the subclasses (`sel.module` is still called below of course)

Reply:
@younik I'm not super comfortable with this change. We've had output checks living at the `gfn/modules.py` for a long time now. Moving that into the estimators does not make sense to me - and seems unrelated to the goal of adding graphs as states into the library.

To me, this really seems like the most abstract place to have such a check. Why do we want to move it out of here? Tagging @saleml for the discussion.

Reply:
I removed because the line `module_output.shape[-1] != self.expected_output_dim()` throw an error when module_output is a tensordict.

Oh, maybe I see it now, are you suggesting to override the `check_output_dim` method in `GraphActionPolicyEstimator` ?

Reply:
Yeah, since this `check_output_dim` check needs to be handled as a special case for `GraphActionPolicyEstimator` -- but I'm open to the discussion.

Reply:
Overriding `check_output_dim`sounds good!